compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -144,6 +144,9 @@ impl<'a, 'tcx> ConstToPat<'a, 'tcx> {
                 traits::NonStructuralMatchTyKind::Foreign => {
                     bug!("use of a value of a foreign type inside a pattern")
                 }
+                traits::NonStructuralMatchTyKind::TyAlias => {
+                    bug!("use of a value of a type alias inside a pattern")
```

Comment:
Is this actually not allowed, or do we expect to normalize past TyAlias by the time we get to building THIR?

Reply:
From my understanding, the latter.

Reply:
Then it might be useful to modify the comment to say something along those lines

Reply:
Done!